docs/core_docs/docs/tutorials/chatbot.ipynb

```diff
@@ -1,1149 +1,1448 @@
+      "    \"tokenUsage\": {\n",
+      "      \"promptTokens\": 58,\n",
+      "      \"completionTokens\": 15,\n",
+      "      \"totalTokens\": 73\n",
+      "    },\n",
+      "    \"finish_reason\": \"stop\",\n",
+      "    \"usage\": {\n",
+      "      \"prompt_tokens\": 58,\n",
+      "      \"completion_tokens\": 15,\n",
+      "      \"total_tokens\": 73,\n",
+      "      \"prompt_tokens_details\": {\n",
+      "        \"cached_tokens\": 0,\n",
+      "        \"audio_tokens\": 0\n",
+      "      },\n",
+      "      \"completion_tokens_details\": {\n",
+      "        \"reasoning_tokens\": 0,\n",
+      "        \"audio_tokens\": 0,\n",
+      "        \"accepted_prediction_tokens\": 0,\n",
+      "        \"rejected_prediction_tokens\": 0\n",
+      "      }\n",
+      "    },\n",
+      "    \"system_fingerprint\": \"fp_bba3c8e70b\"\n",
+      "  },\n",
+      "  \"tool_calls\": [],\n",
+      "  \"invalid_tool_calls\": [],\n",
+      "  \"usage_metadata\": {\n",
+      "    \"output_tokens\": 15,\n",
+      "    \"input_tokens\": 58,\n",
+      "    \"total_tokens\": 73,\n",
+      "    \"input_token_details\": {\n",
+      "      \"audio\": 0,\n",
+      "      \"cache_read\": 0\n",
+      "    },\n",
+      "    \"output_token_details\": {\n",
+      "      \"audio\": 0,\n",
+      "      \"reasoning\": 0\n",
+      "    }\n",
+      "  }\n",
+      "}\n"
+     ]
+    }
+   ],
+   "source": [
+    "const output4 = await app.invoke({ messages: input2 }, config)\n",
+    "console.log(output4.messages[output4.messages.length - 1]);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This is how we can support a chatbot having conversations with many users!\n",
+    "\n",
+    "Right now, all we've done is add a simple persistence layer around the model. We can start to make the more complicated and personalized by adding in a prompt template.\n",
+    "\n",
+    "## Prompt templates\n",
+    "\n",
+    "Prompt Templates help to turn raw user information into a format that the LLM can work with. In this case, the raw user input is just a message, which we are passing to the LLM. Let's now make that a bit more complicated. First, let's add in a system message with some custom instructions (but still taking messages as input). Next, we'll add in more input besides just the messages.\n",
+    "\n",
+    "To add in a system message, we will create a `ChatPromptTemplate`. We will utilize `MessagesPlaceholder` to pass all the messages in."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import { ChatPromptTemplate, MessagesPlaceholder } from \"@langchain/core/prompts\";\n",
+    "\n",
+    "const promptTemplate = ChatPromptTemplate.fromMessages([\n",
+    "  [\"system\", \"You talk like a pirate. Answer all questions to the best of your ability.\"],\n",
+    "  new MessagesPlaceholder(\"messages\"),\n",
```

Comment:
nit: use shorthand:

```
["placeholder", "{messages}"],
```

But also, should we just be using a function rather than a prompt template?